mopidy/stream/actor.py

```diff
@@ -55,3 +62,78 @@ def lookup(self, uri):
             track = Track(uri=uri)
 
         return [track]
+
+
+class StreamPlaybackProvider(backend.PlaybackProvider):
+
+    def __init__(self, audio, backend, config):
+        super(StreamPlaybackProvider, self).__init__(audio, backend)
+        self._config = config
+        self._scanner = backend._scanner
+
+    def translate_uri(self, uri):
+        try:
+            scan_result = self._scanner.scan(uri)
+        except exceptions.ScannerError as e:
+            logger.warning(
+                'Problem scanning URI %s: %s', uri, e)
+            return None
+
+        if not (scan_result.mime.startswith('text/') or
+                scan_result.mime.startswith('application/')):
+            return uri
+
+        content = self._download(uri)
+        if content is None:
+            return None
+
+        tracks = list(playlists.parse(content))
+        if tracks:
+            # TODO Test streams and return first that seems to be playable
```

Comment:
Could also be round-robin, or random. Just so that we play nice and don't give the first item in the list even more traffic by being a badly behaved client.